src/test/java/org/eclipse/basyx/testsuite/regression/aas/factory/aasx/TestMetamodelToAASXConverter.java

```diff
@@ -174,6 +197,61 @@ public void testFilePathsAreCorrectlyChanged() throws IOException, TransformerEx
 		assertFilepathsAreCorrect(aasBundle);
 	}
 	
+	@Test
+	public void testOperationWithVariables() throws Exception {
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
+		MetamodelToAASXConverter.buildAASX(aasList, assetList, conceptDescriptionList, submodelList, fileList, out);
+
+		Set<AASBundle> aasBundles = deserializeAASX(out);
+		assertOperationContainsExpectedVariables(aasBundles);
+	}
+
+	@Test
+	public void testOperationWithoutVariables() throws Exception {
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
+		Operation operation = findOperation(submodelList);
+		operation.setInputVariables(Collections.emptyList());
+		operation.setOutputVariables(Collections.emptyList());
+		operation.setInOutputVariables(Collections.emptyList());
+		MetamodelToAASXConverter.buildAASX(aasList, assetList, conceptDescriptionList, submodelList, fileList, out);
```

Comment:
@FrankSchnicke without this validation, the following error would be thrown. I thought it would make sense to allow submodels with empty operations  
```
java.lang.RuntimeException: Unexpected object: {}
	at org.eclipse.basyx.aas.factory.xml.AASXPackageExplorerCompatibilityHandler.createUnexpectedObjectRuntimeException(AASXPackageExplorerCompatibilityHandler.java:111)
	at org.eclipse.basyx.aas.factory.xml.AASXPackageExplorerCompatibilityHandler.handleInvalidVariableMap(AASXPackageExplorerCompatibilityHandler.java:73)
	at org.eclipse.basyx.aas.factory.xml.AASXPackageExplorerCompatibilityHandler.prepareOperationVariableMap(AASXPackageExplorerCompatibilityHandler.java:62)
	at org.eclipse.basyx.submodel.factory.xml.converters.submodelelement.operation.OperationXMLConverter.parseOperation(OperationXMLConverter.java:81)
	at org.eclipse.basyx.submodel.factory.xml.converters.submodelelement.SubmodelElementXMLConverter.getSubmodelElement(SubmodelElementXMLConverter.java:186)
	at org.eclipse.basyx.submodel.factory.xml.converters.submodelelement.SubmodelElementXMLConverter.getSubmodelElements(SubmodelElementXMLConverter.java:132)
	at org.eclipse.basyx.submodel.factory.xml.converters.submodelelement.SubmodelElementXMLConverter.parseSubmodelElements(SubmodelElementXMLConverter.java:115)
	at org.eclipse.basyx.submodel.factory.xml.converters.SubmodelXMLConverter.parseSubmodels(SubmodelXMLConverter.java:83)
	at org.eclipse.basyx.aas.factory.xml.XMLToMetamodelConverter.<init>(XMLToMetamodelConverter.java:80)
	at org.eclipse.basyx.aas.factory.aasx.AASXToMetamodelConverter.retrieveAasEnv(AASXToMetamodelConverter.java:103)
	at org.eclipse.basyx.aas.factory.aasx.AASXToMetamodelConverter.retrieveAASBundles(AASXToMetamodelConverter.java:118)
```